app/code/core/Mage/Checkout/Block/Onepage/Billing.php

```diff
@@ -102,20 +102,7 @@ public function getMethod()
     public function getAddress()
     {
         if (is_null($this->_address)) {
-            if ($this->isCustomerLoggedIn()) {
-                $this->_address = $this->getQuote()->getBillingAddress();
-                if (!$this->_address->getFirstname()) {
```

Comment:
I would not remove these lines. They've been added somewhere in the past years and serve their purpose: it prefills the name for an address. There's nothing wrong with that.

Reply:
So the suggestion is to 
1. move `$this->_address = $this->getQuote()->getBillingAddress();` above `if ($this->isCustomerLoggedIn()) {`
2. and remove the else?

Reply:
Yes, I think that would make this PR complete.